File: packages/dataframe/rust/stats/distributions/gamma.test.R
#!/usr/bin/env Rscript

# Gamma distribution test script
# Usage: Rscript gamma.test.R <function> <x_or_p> <shape> <rate> [lower_tail] [log_p]

source("test-helpers.R")

# Define gamma distribution functions
gamma_density <- function(x, shape, rate, log = FALSE) {
  dgamma(x, shape, rate, log = log)
}

gamma_cumulative <- function(x, shape, rate, lower.tail = TRUE, log.p = FALSE) {
  pgamma(x, shape, rate, lower.tail = lower.tail, log.p = log.p)
}

gamma_quantile <- function(p, shape, rate, lower.tail = TRUE, log.p = FALSE) {
  qgamma(p, shape, rate, lower.tail = lower.tail, log.p = log.p)
}

gamma_random <- function(n, shape, rate) {
  rgamma(n, shape, rate)
}

# Run the test
run_distribution_test(gamma_density, gamma_cumulative, gamma_quantile, gamma_random)

File: packages/dataframe/rust/stats/distributions/gamma.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust gamma functions...\n");

  const testCases = [
    { func: "dgamma", distribution: "gamma", args: [1, 2, 1, 0] },
    { func: "pgamma", distribution: "gamma", args: [1, 2, 1, 1, 0] },
    { func: "qgamma", distribution: "gamma", args: [0.5, 2, 1, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rgamma separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rgammaResult = await callR("rgamma", "gamma", 1, 2, 1);
    console.log(`rgamma(2, 1) = ${rgammaResult}`);
  } catch (error) {
    console.log(`rgamma: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
